### PR TITLE
mutation to modify shareholder email and address

### PR DESCRIPTION
**Pull request by cballantyne**
Can only be run by user linked to shareholder being updated or by user linked to officeholder for org shareholder belongs to.

### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] If this change fixes a bug (or a performance problem), is a regression
  test (or a benchmark) included?
- [ ] Is a documentation update included (if this change modifies
  existing APIs, or introduces new ones)?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### CODE REVIEW

File: web/web/schema.py [hunk 1]
```diff
@@ -298,11 +340,22 @@ def is_officeholder(organisation_id, user_id):
             return False
 
 
+class ShareholderUtils:
+    @staticmethod
+    def is_shareholder(shareholder, user_id):
```
Comment (ejjick) [inline web/web/schema.py]: This one is a bit confusing. While `is_officeholder` means officeholder of organisation, this one means shareholder entry is owned by user... Should we reserve `is_shareholder` for when it applies to an organisation and rename this one?

Comment (cballantyne) [reply web/web/schema.py]: Good call. Trying to think what could be renamed to though. is_linked_to_shareholder?

Comment (ejjick) [reply web/web/schema.py]: Haha yeah something like that, what I came up with was as verbose... `is_shareholder_owner`

Comment (cballantyne) [reply web/web/schema.py]: going with - is_user_linked_to_shareholder. have refactored and cleaned up some prints in one of the tests and pushed.

**Pull request merged**